invenio_rdm_records/resources/serializers/schemaorg/schema.py

```diff
@@ -0,0 +1,244 @@
+        """Get name identifier. Schemaorg expects a URL for the identifier, and
+        does not support multiple identifiers. Use the first identifier found.
+        """
+
+        def format_name_identifier(name_identifier) -> Optional[str]:
+            """format_name_identifier"""
+            if name_identifier.get("identifier", None) is None:
+                return None
+            if name_identifier.get("scheme", None) == "orcid":
+                return normalize_orcid(name_identifier.get("identifier", None))
+            if name_identifier.get("scheme", None) == "ror":
+                return normalize_ror(name_identifier.get("identifier", None))
+            if name_identifier.get("scheme", None) == "isni":
+                return "http://isni.org/isni/" + name_identifier.get("identifier", None)
+            # TODO: normalize other personororg identifiers, e.g. GND, etc.
+            # Schemaorg expects a URL for the identifier
+            return name_identifier.get("identifier", None)
+
+        return next(
+            (
+                format_name_identifier(i)
+                for i in wrap(obj["person_or_org"].get("identifiers", None))
+            ),
+            None,
+        )
+
+    def get_affiliation(self, obj) -> Optional[list]:
+        """Get affiliation list."""
+
+        def format_affiliation(affiliation) -> Optional[dict]:
+            """format_affiliation"""
+            # TODO: get affiliation id
+            return {"name": affiliation.get("name", None), "@type": "Organization"}
+
+        affiliations = wrap(obj.get("affiliations", None))
+        return [
+            format_affiliation(a) for a in affiliations if a.get("name", None)
+        ] or missing
+
+
+class SchemaorgSchema(BaseSerializerSchema, CommonFieldsMixin):
+    """Schemaorg Marshmallow Schema."""
+
+    include = {
+        "@context": fields.Constant("http://schema.org"),
+        "@id": fields.Method("get_id"),
+        "@type": fields.Method("get_type"),
+    }
+
+    identifier = fields.Method("get_identifiers")
+    name = SanitizedUnicode(attribute="metadata.title")
+    author = fields.List(
+        fields.Nested(PersonOrOrgSchema), attribute="metadata.creators"
+    )
+    editor = fields.List(
+        fields.Nested(PersonOrOrgSchema), attribute="metadata.contributors"
+    )
+    publisher = fields.Method("get_publisher")
+    keywords = fields.Method("get_keywords")
+    datePublished = fields.Method("get_publication_date")
+    dateModified = fields.Method("get_modification_date")
+    inLanguage = fields.Method("get_language")
+    contentSize = fields.Method("get_size")
+    encodingFormat = fields.Method("get_format")
+    version = SanitizedUnicode(attribute="metadata.version")
+    license = fields.Method("get_license")
+    description = SanitizedUnicode(attribute="metadata.description")
+    # spatialCoverage = fields.Method("get_spatial_coverage")
+    # funder = fields.Method("get_funder")
```

Comment:
Any particular reason why `funding` was omitted? If just needs implementation, I can take over and implement it :)

Reply:
No particular reason, just lack of time regarding the original deadline for the Zenodo relaunch. I can add funding in a future iteration.